src/datapoints-graph/datapoints-graph-config/datapoints-graph-widget-config.component.ts

```diff
@@ -64,16 +68,22 @@ export class DatapointsGraphWidgetConfigComponent
     showRange: true,
     showChart: true,
   };
+  datapointSelectionConfig: Partial<DatapointSelectorModalOptions> = {};
   activeDatapointsExists: boolean;
   private destroy$ = new Subject<void>();
 
   constructor(
     private formBuilder: FormBuilder,
     private form: NgForm,
-    private translate: TranslateService
+    private translate: TranslateService,
+    private route: ActivatedRoute
   ) {}
 
   ngOnInit() {
+    const context = this.route.root.firstChild.snapshot.data?.contextData;
```

Comment:
What about getContextData from the packages/ngx-components/core/router/context-route.service.ts? It does not work? Maybe we could adjust it to support your usecase?

Reply:
getContextData is actually just unpacking provided ActivatedRoute, but not in the way we want to. 
main issue here is that datapoints widget config component is outside router outlet (that's because it is projected to modal that is not in current router outlet, it's placed in <body>). that's why we have to search deeper, in root -> firstChild. so yeah, we can adjust getContextData but it will do the same as I did above, but we would allow to use this feature since 1016.0.259 (we allow to use whole plugin since 1016.0.214), so I'm nnot convinced that it's better way.

Reply:
nah, lets leave it :D